src/NServiceBus.Core/Pipeline/PipelineCache.cs

```diff
@@ -32,7 +32,7 @@ public IPipeline<TContext> Pipeline<TContext>()
             {
                 return (IPipeline<TContext>) lazyPipeline.Value;
             }
-            return default(IPipeline<TContext>);
+            throw new InvalidOperationException("Registration of custom pipelines is not supported.");
```

Comment:
also, I'm not sure about the word "registration" at this place. The sentence is correct but this is thrown when we try to resolve a custom pipeline.